Expensify/App
# [No QA] Change reportID type from number to string

### Fixed Issues
Part of https://github.com/Expensify/Expensify/issues/227425
- [x] HOLD on Web-E PR https://github.com/Expensify/Web-Expensify/pull/34976

### Tests
1. Select a chat in the LHN
2. Verify that the row in the LHN shows that the chat is selected (it has a slightly darker background than the rest)

There are a few propType warnings that happen regarding the IOU chat reportIDs, but that API is not fully migrated over to an OfflineFirst world, so those will be fixed later when it's migrated.


### QA Steps
1. Select a chat in the LHN
2. Verify that the row in the LHN shows that the chat is selected (it has a slightly darker background than the rest)
